lib/src/chatlist/chat_list.dart

```diff
@@ -166,14 +175,99 @@ class _ChatListState extends State<ChatList> {
 
   @override
   Widget build(BuildContext context) {
-    return BlocListener<BackgroundBloc, BackgroundState>(
-      listener: (context, state) {
-        if (state is BackgroundStateSuccess) {
-          if (state.state == AppLifecycleState.resumed.toString()) {
-            shareBloc.add(LoadSharedData());
-          }
-        }
-      },
+    return MultiBlocListener(
+      listeners: [
+        BlocListener<BackgroundBloc, BackgroundState>(
+          listener: (context, state) {
+            if (state is BackgroundStateSuccess) {
+              if (state.state == AppLifecycleState.resumed.toString()) {
+                shareBloc.add(LoadSharedData());
+                _inviteBloc.add(HandleSharedInviteLink());
+              }
+            }
+          },
+        ),
+        BlocListener<InviteBloc, InviteState>(
+          bloc: _inviteBloc,
+          listener: (context, state) {
+            if (state is InviteStateSuccess) {
+              if (state.inviteServiceResponse != null) {
+                InviteServiceResponse inviteServiceResponse = state.inviteServiceResponse;
+                String name = inviteServiceResponse.sender.name;
+                String email = inviteServiceResponse.sender.email;
+                int startIndex = inviteServiceResponse.sender.image.lastIndexOf(',') + 1;
```

Comment:
Please move this handling (getting the index, extracting the actual image) to the BloC.

Reply:
I saw multiple occurrences of this index handling to get the actual data, could that be refactored in a helper class?

Reply:
Is moved to BloC.